integration-tests/test_client_options.py

```diff
@@ -145,3 +145,36 @@ def test_client_diagnosis(insights_client):
     diagnosis_data = json.loads(diagnosis_result.stdout)
     # verify that diagnosis contains correct machine id
     assert diagnosis_data["insights_id"] == machine_id
+
+
+def test_check_show_results(insights_client):
+    """
+    Verify a remediation is returned when checking results from server (--check-results)
+    and then displaying then (--show-results)
+    First I will introduce a vulnerability to the system and at the
+    end of the test I will set the correct file permission again
+    """
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    try:
+        os.chmod("/etc/ssh/sshd_config", 0o777)
+
```

Comment:
So when you introduce vulnerability in system we also need to make inventory aware about this issue. We can do so by running insights-client to upload the fresh data and then we run --check-results to verify that correct data is uploded.